src/components/Navbar.js

```diff
@@ -0,0 +1,178 @@
+import React, { useState, useRef } from 'react';
+import { useAuth } from '../contexts/AuthContext';
+import { useTheme } from '../contexts/ThemeContext';
+import { useKanban } from '../hooks/useKanban';
+
+export function Navbar() {
+  const { user, signInWithGoogle, signOut } = useAuth();
+  const { isDark, toggleTheme } = useTheme();
+  const { boardTitle, updateBoardTitle, addTask, updateColumn, columns } = useKanban();
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(boardTitle);
+  const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
+  const [showForms, setShowForms] = useState(false);
+  const titleInputRef = useRef(null);
+
+  const handleTitleClick = () => {
+    if (user) {
+      setIsEditingTitle(true);
+      setEditedTitle(boardTitle);
+    }
+  };
+
+  const handleTitleChange = (e) => {
+    setEditedTitle(e.target.value);
+  };
+
+  const handleTitleBlur = async () => {
+    setIsEditingTitle(false);
+    if (editedTitle.trim() && editedTitle !== boardTitle) {
+      await updateBoardTitle(editedTitle.trim());
+    } else {
+      setEditedTitle(boardTitle);
+    }
+  };
+
+  const handleTitleKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      handleTitleBlur();
+    }
+  };
+
+  const handleAddTask = async (e) => {
+    e.preventDefault();
+    if (!user || !newTask.trim()) return;
+
+    const task = {
+id: crypto.randomUUID(),
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
id: (() => {
  try {
    return crypto.randomUUID();
  } catch (e) {
    return `${Date.now()}-${Math.random().toString(36).substr(2, 9)}`;
  }
})(),
```
Consider wrapping the crypto.randomUUID() call in a try-catch block to handle potential runtime errors that might occur in certain environments or if the crypto API is disabled.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;